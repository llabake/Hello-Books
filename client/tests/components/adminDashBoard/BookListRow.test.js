import React from 'react';

import {
  BookListRow,
  mapDispatchToProps,
  mapStateToProps
} from '../../../components/adminDashBoard/BookListRow';
import adminData from '../../mocks/userData';

function setup() {
  const props = {
    dispatch: jest.fn(),
    user: adminData.admin,
    handleEdit: jest.fn(),
    handleDelete: jest.fn(),
    deleteBookAction: jest.fn(),
    errors: {},
    book: {
      aboutAuthor: 'Great Author',
      author: 'Imaseun Egbosa',
      borrowCount: 0, 
      createdAt: "2018-06-07T15:08:48.835Z",
      description: 'Youth juvenile deliquency',
      downVotes: 0,
      favorited: [],
      id: 5,
      image: '',
      isbn: '7698102987364',
      publishedYear: '2012',
      quantity: 7,
      reviews: [],
      title: 'Fine Boys',
      upVotes: 0,
      updatedAt: '2018-06-07T15:08:48.835Z',
      votes: []
    },
  };

  const wrapper = shallow(<BookListRow {...props} />);

  return {
    wrapper,
    props,
  };
}
describe('Book list row', () => {
  test('should render correctely', () => {
    const { wrapper, props } = setup();
    expect(wrapper).toMatchSnapshot();
  })

  test('mapDispatchToProps', () => {
    const dispatch = jest.fn();

    expect(mapDispatchToProps(dispatch)).toHaveProperty('deleteBookAction');

    const { deleteBookAction } = mapDispatchToProps(dispatch);
    deleteBookAction();
    expect(dispatch).toHaveBeenCalled();
  });

  test('should handle book delete action', () => {
    const { wrapper, props, state } = setup();
    const event = {
      preventDefault: jest.fn(),
      target: {}
    }
    wrapper.instance().handleEdit(event)
    // expect(props.handleEdit).toHaveBeenCalled()
  });

  test('should handle book edit action', () => {
    const { wrapper, props, state } = setup();
    const event = {
      preventDefault: jest.fn(),
      target: {}
    }
    wrapper.instance().handleDelete(event)
    // expect(props.handleDelete).toHaveBeenCalled()
  });
  test('mapStateToProps', () => {
    expect(mapStateToProps({ adminReducer: {} })).toHaveProperty('errors');
    expect(mapStateToProps({ adminReducer: {} })).toHaveProperty('allBooks');
  });
})