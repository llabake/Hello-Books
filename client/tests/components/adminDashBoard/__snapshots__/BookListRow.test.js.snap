// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book list row should render correctely 1`] = `
<tr>
  <td>
    NaN
  </td>
  <td>
    Fine Boys
  </td>
  <td>
    Imaseun Egbosa
  </td>
  <td>
    <a
      className="modal-trigger .modal-close"
      id="edit-book"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <i
        className="material-icons"
      >
        edit
      </i>
    </a>
  </td>
  <td>
    <a
      id="delete-book"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <i
        className="material-icons"
      >
        delete
      </i>
    </a>
  </td>
</tr>
`;
